mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -147,14 +150,63 @@ public RunInfo createRun(CreateRun request) {
    * @param runViewType ViewType for expected runs. One of (ACTIVE_ONLY, DELETED_ONLY, ALL)
    *                    Defaults to ACTIVE_ONLY.
    *
+   * @return A list of all RunInfos that satisfy search filter.
+   */
+  public List<RunInfo> searchRuns(List<String> experimentIds,
+                              String searchFilter,
+                              ViewType runViewType) {
+    SearchRuns.Builder builder = SearchRuns.newBuilder()
+            .addAllExperimentIds(experimentIds);
+
+    if (searchFilter != null) {
+      builder.setFilter(searchFilter);
+    }
+    if (runViewType != null) {
+      builder.setRunViewType(runViewType);
+    }
+    SearchRuns request = builder.build();
+    String ijson = mapper.toJson(request);
+    String ojson = sendPost("runs/search", ijson);
+    return mapper.toSearchRunsResponse(ojson).getRunsList().stream().map(Run::getInfo)
+            .collect(Collectors.toList());
+  }
+
+  /**
+   * Return runs from provided list of experiments that satisfy the search query.
+   *
+   * @param experimentIds List of experiment IDs.
+   * @param searchFilter SQL compatible search query string. Format of this query string is
+   *                     similar to that specified on MLflow UI.
+   *                     Example : "params.model = 'LogisticRegression' and metrics.acc != 0.9"
+   * @param maxResults Maximum number of runs desired. Defaults to 1000.
+   *
    * @return A list of all Runs that satisfy search filter.
    */
-  public List<Run> searchRuns(List<String> experimentIds,
+  public RunsPage searchRuns(List<String> experimentIds,
                                   String searchFilter,
-                                  ViewType runViewType) {
-    return searchRuns(experimentIds, searchFilter, runViewType, new ArrayList<>());
+                                  int maxResults) {
+    return searchRuns(experimentIds, searchFilter, ViewType.ACTIVE_ONLY, new ArrayList<>(),
```

Comment:
this will call the wrapper 
```
  public RunsPage searchRuns(List<String> experimentIds,
                              String searchFilter,
                              ViewType runViewType,
                              List<String> orderBy,
                              int maxResults)  { 
  ...
}
```
instead of the one that calls that calls the REST endpoint.

Reply:
So we have 
```
// v1
searchRuns(experimentIds, searchFilter)
searchRuns(experimentIds, searchFilter, runViewType)
// v2 
searchRuns(experimentIds, searchFilter, maxResults)
searchRuns(experimentIds, searchFilter, orderBy, maxResults)
searchRuns(experimentIds, searchFilter, runViewType, orderBy, maxResults) 
searchRuns(experimentIds, searchFilter, runViewType, orderBy, maxResults, token) 
```
This seems pretty confusing...

I've usually seen java APIs do something like
```
func(a)
func(a, b)
func(a, b, c)
```
never seen
```
func(c)
func(b, c)
func(a, b, c)
```

@mparkhe @aarondav what do you folks think?

Reply:
I think, we should be ok in this case, since we are deprecating `v1` methods.
Also the `v2` methods have very different signatures -- even different # of arguments.

Reply:
What about the ordering of the arguments for the `v2` methods?

Reply:
Yes this is something I wanted feedback on. It seems annoying for the user to have to specify `runViewType` and `orderBy` as the defaults in order to specify `maxResults`. It's nice in python that the defaults can just get loaded in. 

But in Java having different permutations of the arguments seems weird. I can take out the `searchRuns(experimentIds, searchFilter, maxResults)` which is the only one out of order. @mparkhe  @aarondav

Reply:
Yeah we definitely need to keep the order consistent within the v2 ones. We could also consider requiring runViewType if you want orderBy to keep them consistent with v1.

Reply:
You could have these `v2` methods

```
searchRuns(experimentIds, searchFilter, runViewType, maxResults)
searchRuns(experimentIds, searchFilter, runViewType, maxResults, orderBy)
searchRuns(experimentIds, searchFilter, runViewType, maxResults, orderBy, token) 
```

Reply:
I also like that ordering better than orderBy first.